### PR TITLE
Book Store Connect to API

### PR DESCRIPTION
**Pull request by SaboorSohaib**
### Hi Dear!👋👋👋

**I implemented the following requirements.**

- Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to 
  learn how to use the API.
- Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios- 
  http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
- Fetch book list from Bookstore API on Books page load.
  Add an Action Creator, that returns a function.
  Fetch a list of books.
  Dispatch another action (add an action type) that will pass the API response to the reducer and update the state.
- Refactor your add book and remove book features to persist your changes in the server.
  Change your existing Action Creators, so they return functions.
  Make changes by using API.
  Dispatch an action (action type that existed before refactor) that will update the state accordingly.
- Add a book and remove book should work in the same way after the refactor.
- No styling is required.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,50 +1,71 @@
...
-  {
-    id: 1,
-    title: 'Advance React',
-    author: 'Mosh',
-  },
-  {
-    id: 2,
-    title: 'Advance Redux',
-    author: 'Abdul Saboor',
+/* eslint-disable no-param-reassign */
```
Comment (emmiiorji) [inline src/redux/books/books.js:1]: As much as possible, kindly avoid disabling linter checks in line. They represent coding standards you share with your team and most of the time enforce best coding practices.

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+// import CategoriesReducer from './categories/categories';
+// import books from './books/books';
+
+// const rootReducer = combineReducers({
+//   CategoriesReducer,
+//   books,
+// });
+
+// const store = configureStore({ reducer: rootReducer });
+// export default store;
```
Comment (emmiiorji) [inline src/redux/configureStore.js]: Kindly remove unused pieces of code in order to clean up the code base.

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

A little bit more

## Good Highlights✔️

- Linter checks are passing
- Good commit messages
- API requests made properly


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @SaboorSohaib, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- All API requests have been made correctly :+1: 
- PR is descriptive :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**